app/src/bcsc-theme/features/modal/DeviceInvalidated.tsx

```diff
@@ -0,0 +1,35 @@
+import { useFactoryReset } from '@/bcsc-theme/api/hooks/useFactoryReset'
+import { TOKENS, useServices } from '@bifold/core'
+import { useCallback } from 'react'
+import { SystemModal } from './components/SystemModal'
+
+/**
+ * Component displayed when the device has been invalidated.
+ *
+ * @returns {*} {JSX.Element} The DeviceInvalidated component.
+ */
+export const DeviceInvalidated = (): JSX.Element => {
+  const factoryReset = useFactoryReset()
+  const [logger] = useServices([TOKENS.UTIL_LOGGER])
+
+  /**
+   * Handles the factory reset operation.
+   */
+  const handleFactoryReset = useCallback(async () => {
+    const result = await factoryReset()
+
+    if (!result.success) {
+      logger.error('Factory reset failed', result.error)
+    }
+  }, [factoryReset, logger])
```

Comment:
context: I have some stuff in my PR that updates the useFactoryReset hook to allow a partial state object to be passed.

Without it, calling factory reset will fully reset the state including the flag for onboarding.

Depends if you need the user to see the onboarding screens after they call this.